color.d

```diff
@@ -1777,14 +1779,45 @@ struct Point {
 	Point opBinary(string op)(int rhs) @nogc {
 		return Point(mixin("x" ~ op ~ "rhs"), mixin("y" ~ op ~ "rhs"));
 	}
+
+	Size opCast(T = Size)() inout @nogc {
+		return Size(x, y);
+	}
 }
 
 ///
 struct Size {
 	int width; ///
 	int height; ///
 
-	int area() pure nothrow @safe const @nogc { return width * height; }
+	pure nothrow @safe:
+
+	int area() const @nogc { return width * height; }
+
+	int offset(const int x, const int y) const @nogc {
+		return linearOffset(width, Point(x, y));
+	}
+
+	int offset(const Point p) const @nogc {
+		return linearOffset(width, p);
+	}
+
+	Point opCast(T = Point)() inout @nogc {
+		return Point(width, height);
+	}
```

Comment:
I guess, we’d want to have doc comments for them. Is there a particular reason why `area()` hasn’t got one yet?

Reply:
I added it for just a quick little hack but it is useful enough and obviously easy math so prolly not buggy (right...? lol) i might as well go ahead and formally announce it

Reply:
doc comment added